lib/core/api.js

```diff
@@ -636,6 +657,23 @@ api.getMostRecentUploadRecord = function(userId, deviceId, cb) {
   });
 };
 
+api.upload.blob = function(blob, contentType) {
+  api.log('POST /blobs');
+
+  var digest = btoa(md5(blob, null, true));
+  var blobObject = new Blob([blob], {type: contentType});
+
+  return new Promise((resolve, reject) => {
```

Comment:
Basically every other `api.` exposed function follows the function signature of accepting a callback as the last argument as opposed to returning a `Promise`. I won't argue that the the newer `Promise` symantics aren't better (oh, they're so much better...) but I don't think we should introduce the new syntax one function at a time as it'll lead to a lack of internal consistency across the api surface. (Does this function return a `Promise` vs. do I give it a callback.) I'd love to make a sweeping set of changes to update a lot of this sort of thing, but I'm going to recommend sticking with the callback for now.